client/repositories/signature/signature.go

```diff
@@ -0,0 +1,104 @@
+package signature_repo
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/lidofinance/dc4bc/client/modules/state"
+	"github.com/lidofinance/dc4bc/client/types"
+)
+
+const (
+	SignaturesKeyPrefix = "signatures"
+)
+
+type SignatureRepo interface {
+	SaveSignatures(signature []types.ReconstructedSignature) error
+	GetSignatureByID(dkgID, signatureID string) ([]types.ReconstructedSignature, error)
+	GetSignatures(dkgID string) (map[string][]types.ReconstructedSignature, error)
+}
+
+type BaseSignatureRepo struct {
+	state state.State
+}
+
+func NewSignatureRepo(state state.State) *BaseSignatureRepo {
+	return &BaseSignatureRepo{state}
+}
+
+func (r *BaseSignatureRepo) GetSignatures(dkgID string) (signatures map[string][]types.ReconstructedSignature, err error) {
+	key := state.MakeCompositeKey(SignaturesKeyPrefix, dkgID)
```

Comment:
since you're casting the key to a string later, it's better to use the MakeCompositeKeyString you've defined